iroh-net/src/disco.rs

```diff
@@ -25,9 +25,9 @@ use std::{
 
 use anyhow::{anyhow, ensure, Result};
 
-use crate::net::ip::to_canonical;
+use crate::{key, net::ip::to_canonical};
 
-use super::{key, stun};
+use super::{key::PublicKey, stun};
```

Comment:
this `super` is a bit weird, importing from both super::key and crate::key which is the same module.  I'd say move everything to `crate::`.